app/src/main/java/com/greenfox/kalendaryo/fragments/AccountsFragment.java

```diff
@@ -5,37 +5,51 @@
 import android.support.annotation.NonNull;
 import android.support.design.widget.FloatingActionButton;
 import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ListView;
 
 import com.google.android.gms.common.ConnectionResult;
 import com.google.android.gms.common.api.GoogleApiClient;
 import com.greenfox.kalendaryo.LoginActivity;
 import com.greenfox.kalendaryo.R;
+import com.greenfox.kalendaryo.adapter.AccountAdapter;
 import com.greenfox.kalendaryo.adapter.AccountsList;
-import com.greenfox.kalendaryo.models.GoogleAuth;
+import com.greenfox.kalendaryo.components.DaggerApiComponent;
+import com.greenfox.kalendaryo.http.backend.BackendApi;
 import com.greenfox.kalendaryo.models.KalPref;
+import com.greenfox.kalendaryo.models.responses.AuthResponse;
 
-import java.util.List;
+import javax.inject.Inject;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class AccountsFragment extends Fragment implements GoogleApiClient.OnConnectionFailedListener {
 
-    ListView accountNamesView;
     AccountsList adapter;
+    AccountAdapter accountAdapter;
     KalPref kalpref;
     FloatingActionButton floatingActionButton;
+    RecyclerView recyclerView;
+
+    @Inject
+    BackendApi backendApi;
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
-        View view = inflater.inflate(R.layout.listaccounts, container, false);
+        View view = inflater.inflate(R.layout.listaccounts_recycler, container, false);
         kalpref = new KalPref(getActivity());
-        adapter = new AccountsList(view.getContext());
-        accountNamesView = view.findViewById(R.id.allaccounts);
-        accountNamesView.setAdapter(adapter);
+        DaggerApiComponent.builder().build().inject(this);
+        recyclerView = view.findViewById(R.id.accountRecyclerView);
+        LinearLayoutManager recyclerLayoutManager = new LinearLayoutManager(this.getContext());
+        recyclerView.setLayoutManager(recyclerLayoutManager);
         floatingActionButton = view.findViewById(R.id.addNewAccount);
+        getAuthResponse(kalpref.clientToken());
```

Comment:
getAuthResponse() logic is added here and in `ChooseAccountActivity.java` too
maybe could refactor and move the logic into a service?